From 64c70fcdd9751cf1e8d25f1240fa7a0ea4e353fc Mon Sep 17 00:00:00 2001
From: Mark Wielaard <mark@klomp.org>
Date: Wed, 26 Jul 2017 02:37:17 +0200
Subject: [PATCH] Make sure that debugsourcefiles.list is generated in the
 build dir.

The %_debugsource_template expects the debugsourcefiles.list file
to be in the (current) build dir. Make sure that is always the case
even if find-debuginfo.sh was invoked in a different (sub) directory
by adding the build dir path as explicit argument to -S.

Signed-off-by: Mark Wielaard <mark@klomp.org>
---
 macros.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/macros.in b/macros.in
index f71fcc103..d882fa794 100644
--- a/macros.in
+++ b/macros.in
@@ -185,7 +185,7 @@
     %{?_unique_debug_srcs:--unique-debug-src-base "%{name}-%{VERSION}-%{RELEASE}.%{_arch}"} \\\
     %{?_find_debuginfo_dwz_opts} \\\
     %{?_find_debuginfo_opts} \\\
-    %{?_debugsource_packages:-S debugsourcefiles.list} \\\
+    %{?_debugsource_packages:-S "%{_builddir}/%{?buildsubdir}/debugsourcefiles.list"} \\\
     "%{_builddir}/%{?buildsubdir}"\
 %{nil}
 
-- 
2.13.3

